### PR TITLE
[WIP] mgr/dashboard: Centralized services for url / links from source code for easy iteration and unit-testing

### PR DESCRIPTION
**Pull request by Emmanuel1999**
Signed-off-by: Emmanuel Opatola <emmanuelopatola99@gmail.com>
Fixes: https://tracker.ceph.com/issues/47338 #46002

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/tsconfig.json [hunk 1]
```diff
@@ -19,13 +19,13 @@
...
-    "baseUrl": "./",
+    "baseUrl": "src",
     "paths": {
       "~/*": ["src/*"]
     },
-    "typeRoots": [
+    /*"typeRoots": [
       "node_modules/@types"
-    ],
+    ],*/
```
Comment (pereman2) [inline src/pybind/mgr/dashboard/frontend/tsconfig.json:28]: why

**Review by pereman2 — COMMENTED** [review]
Isn't e2e or unittest missing from this PR?

**Comment by ljflores** [comment]
@pereman2 I marked this as a WIP since @Emmanuel1999 is still working on it.

**Comment by pereman2** [comment]
> @pereman2 I marked this as a WIP since @Emmanuel1999 is still working on it.

@ljflores Great! Thank you

**Comment by ljflores** [comment]
@Emmanuel1999 can we help with this PR at all?

**Comment by Emmanuel1999** [comment]
I mistakenly closed this pull request when I attempted removing my last commit, Please, I've opened another pull request https://github.com/ceph/ceph/pull/46177 please kindly review and provide feedback.

**Comment by Emmanuel1999** [comment]
> 



> @Emmanuel1999 can we help with this PR at all?

When I tried pulling my initial commit, the pull request closed also, so I opened a new one.

**Pull request closed without merging**